### PR TITLE
limit nodes and reduce CI jobs

### PR DESCRIPTION
**Pull request by cofyc**
Signed-off-by: Yecheng Fu <fuyecheng@pingcap.com>

### What problem does this PR solve? 

fixes https://github.com/pingcap/tidb-operator/issues/1603

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: ci/pingcap_tidb_operator_build_kind.groovy [hunk 1]
```diff
@@ -190,9 +227,6 @@ def call(BUILD_BRANCH, CREDENTIALS_ID, CODECOV_CREDENTIALS_ID) {
 		builds["E2E v1.12.10"] = {
 			build("${MIRRORS} IMAGE_TAG=${GITHASH} SKIP_BUILD=y GINKGO_NODES=8 KUBE_VERSION=v1.12.10 REPORT_DIR=\$(pwd)/artifacts REPORT_PREFIX=v1.12.10_ ./hack/e2e.sh -- --preload-images --ginkgo.skip='\\[Serial\\]'", artifacts)
 		}
-		builds["E2E v1.16.4"] = {
```
Comment (weekface) [inline ci/pingcap_tidb_operator_build_kind.groovy]: remove k8 1.16?

Comment (cofyc) [reply ci/pingcap_tidb_operator_build_kind.groovy]: yes, reduce resource needed for our CI jobs.
if we test against 1.16 on PR, why not 1.15, 1.14, ...
before we have enough resources to use, we can test only the minimum and the latest versions on PR.
of cause, we can have a job to test against all versions on branch serially.

Comment (cofyc) [reply ci/pingcap_tidb_operator_build_kind.groovy]: xref: https://github.com/pingcap/tidb-operator/issues/1616

**Pull request merged**